### ISSUE TITLE
Implementation pipe flow

### ISSUE THREAD

**Issue by rvicentecruz** [xcompact3d/Incompact3d#178]
Hello, 

As discussed during the last showcase event (27-28 April 2023, Imperial College London), it would be interesting to implement the pipe flow into the main branch of the code, given its academic nature and importance among the canonical wall-bounded flows. Through my thesis, I have worked on the full implementation of this type of flow in Xcompact3d while including heat transfer with the pipe geometry being represented by the Lagrange polynomial immersed boundary method `iibm=2`. 

I started the implementation of the pipe flow in 2019, starting from the version of the code available in the main branch at that time. As the ultimate goal of [my thesis](https://hal.science/tel-03605404/) was the introduction of a IB-based numerical strategy for Conjugate Heat Transfer simulations, [many developments](https://github.com/rvicentecruz/Xcompact3d_Pipe_CHT/blob/main/src/BC-Pipe-flow.f90) were made throughout these 4 years, including [viscous filtering](https://doi.org/10.1016/j.jcp.2021.110115) and fluid-solid thermal coupling for complex geometry. This [version of the code](https://github.com/rvicentecruz/Xcompact3d_Pipe_CHT) is fully operational and I believe it can serve as a basis to guide the implementation in the main branch.

As we discussed during the showcase event, perhaps we could start with the basic implementation/validation of the velocity solution alone as a first step (geometry creation in the frame of IBM, laminar/turbulent initial conditions, ...) before moving on to the heat transfer implementation.  Naturally, I'm willing to restart from the basics and advance step-by-step, according to the guidelines you provide to implement this flow configuration. 

Kind regards,
Rodrigo Vicente Cruz

**Comment by pbartholomew08** [comment]
Hi Rodrigo,

I think as you suggest that doing this as a series of steps will be best all-round. A pull request for the pipeflow geometry and flow conditions would I think be a good starting point. We can then build on this from there.

Cheers,
Paul

**Comment by rvicentecruz** [comment]
Hello Paul,

Thank you for your reply, I will prepare this and submit a pull request as soon as possible.

Cheers,
Rodrigo

**Comment by pbartholomew08** [comment]
Hi Rodrigo, you can start a draft pull request - it doesn't necessarily need to be finished before doing so, this helps others to see what is going on and either contribute or comment, ultimately this should make it easier to get merged into the code.

Cheers,
Paul

**Comment by tlestang** [comment]
A draft PR is useful but I would suggest starting with opening another issue going into a bit more technical details in describing the plan for that first step. It would useful to discuss what will have to be added or modified, so that we can all get a feel for the extent of the changes and be aware of your plan @rvicentecruz .

**Comment by rvicentecruz** [comment]
Hello @pbartholomew08 and @tlestang, 

Thank you both for your replies. So, it would be a new issue listing only the technical guidelines for the implementation of this first step (geometry + initial conditions) ? As I understand it, when opening the new issue, I should not submit pull requests at this time, but simply provide a full description of what would be included in the `BC-Pipe-Flow.f90` file in the first instance, is that correct?

**Comment by tlestang** [comment]
If this first step amounts to including a single new file to the repo - feel free to open a PR for it now. I you expect you'll have to make changes to parts of the code elsewhere in addition to adding the file, it could help to discuss it in an issue first. No hard and fast rules on how exactly to proceed - your call ;)
